parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/AbstractParallelEoSStreamProcessor.java

```diff
@@ -1132,79 +1199,61 @@ private void updateLastCommitCheckTime() {
         lastCommitCheckTime = Instant.now();
     }
 
+    private io.micrometer.core.instrument.Timer userProcessingTimer;
+
+
     /**
-     * Run the supplied function.
+     * Gather metrics from the {@link ParallelConsumer} and for monitoring.
      */
-    protected <R> List<ParallelConsumer.Tuple<ConsumerRecord<K, V>, R>> runUserFunction(Function<PollContextInternal<K, V>, List<R>> usersFunction,
-                                                                                        Consumer<R> callback,
-                                                                                        List<WorkContainer<K, V>> workContainerBatch) {
-        // call the user's function
-        List<R> resultsFromUserFunction;
-        PollContextInternal<K, V> context = new PollContextInternal<>(workContainerBatch);
+    public PCMetrics calculateMetrics() {
```

Comment:
does it make sense to have this manual call still? what should our "health" end point look like?